esmvaltool/preprocessor/_time_area.py

```diff
@@ -112,15 +159,185 @@ def area_average(mycube, coord1, coord2):
     usually 'longitude' and 'latitude' but depends on the cube);
     Returns a cube
     """
+
     import iris.analysis.cartography
-    mycube.coord(coord1).guess_bounds()
-    mycube.coord(coord2).guess_bounds()
+    # CMOR ised data should already have bounds?
+    #    mycube.coord(coord1).guess_bounds()
+    #    mycube.coord(coord2).guess_bounds()
     grid_areas = iris.analysis.cartography.area_weights(mycube)
```

Comment:
Do I understand it right that you guessing area weights? For curvilinear grids, it's not the best strategy since the error might be quite substantial. I think it is much clearer to use `fx` files, provided by modelling groups,  in particular, `areacello`. The information about the location of this files is going to be available  when this issue is solved #423

Reply:
I believe that iris does attempt to guess the area weights from the grid boundaries in the coordinates. 

If there is a method in ESMValTool to do that from the provided area files, then that would be an improvement. 

Better still would be for iris to load the fx files alongside the cube, so that the methods in cartograhpy like area_weights work as intended.

Reply:
Maybe you could open an issue on the [iris github page](https://github.com/SciTools/iris/issues) to request such a feature.

Reply:
At the end of the day, I would compare results that you get by using `area_weights` and by applying the `areacello` directly for some particularly crazy grids. If you are happy with the result, probably it is safe to leave it as is for now.

Reply:
Back to @bouweandela's original comment:
>Specifically (see the Codacy issues, click the Details link below behind Codacy)

I feel pretty silly sending in regular work in progress reports trying to figure out what codacy wants from me. It is possible to run that locally, or is it hard to set up? the yapf/isort utilities don't seem to agree with the code standards used in codacy. 

Also, with regards to the codacy report, do I need to address all the issues, or just the yellow or higher issues? Ie, I can't make codacy able to import numpy (blue warning)!

Reply:
You can run prospector (https://prospector.landscape.io/en/master/) locally as Codacy issues are mostly based on prospector warnings. This greatly speed up the process 

I think you can install it trough conda easily

Reply:
either that or ```sudo apt-get install``` if on Debian; from a low level point of view I'd add a check if the data already has bounds, ```iris``` throws an error if it already has

Reply:
@ledm Indeed it is recommended that you run prospector locally, as described in our README.md file: https://github.com/esmvalgroup/esmvaltool/tree/version2_development#code-style